lib/storage.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const data = {};
+const Promise = require('bluebird');
+const storage = module.exports = {};
+const dataDir = `${__dirname}/../data`;
+const fs = Promise.promisifyAll(require('fs'));
+
+storage.setItem = function(name, item) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    err.status = 400;
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(item);
+    return fs.writeFileAsync(`${dataDir}/${name}/${item.id}.json`, json);
+  })
+  .then(() => {
+    return item;
+  });
+};
+
+storage.getItem = function(name, id) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(id);
+    return fs.readFileAsync(`${dataDir}/${name}/${id}.json`, json);
+  })
+  .then(() => {
+    return id;
+  });
+};
+
+storage.deleteItem = function(name, id) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(id);
+    return fs.readFileAsync(`${dataDir}/${name}/${id}.json`, json);
+  })
+  .then(() => {
+    delete id;
```

Comment:
This is breaking your tests. We'll no longer use the delete method, and instead, you have to research the proper method you need to delete a file from the file system. [Research the Node docs](https://nodejs.org/dist/latest-v6.x/docs/api/fs.html) and CTRL+f for "delete" and try to use that method here and follow the same pattern above.